pkg/catalog/list.go

```diff
@@ -40,10 +41,22 @@ func (m *Manager) List(k8sClient runtimeclient.Client, catalogClient CatalogClie
 		if len(versions) == 0 {
 			versions = append(versions, "-")
 		}
-		profileData = append(profileData, ProfileData{
-			Profile:                 p,
-			AvailableVersionUpdates: versions,
-		})
+
+		// filter results if name is provided
+		if name != "" {
+			if strings.Contains(p.Name, name) {
+				profileData = append(profileData, ProfileData{
+					Profile:                 p,
+					AvailableVersionUpdates: versions,
+				})
+			}
+		} else {
+			profileData = append(profileData, ProfileData{
+				Profile:                 p,
+				AvailableVersionUpdates: versions,
+			})
+		}
```

Comment:
do we wanna unit test this?

Reply:
yes i am adding few more tests around this